File: src/algorithm/leetcode/Problem226.java
package algorithm.leetcode;

public class Problem226 {

    public static TreeNode invertTree(TreeNode root) {
        if (root != null) {
            TreeNode left = root.left;
            root.left = invertTree(root.right);
            root.right = invertTree(left);
        }
        return root;
    }
}

File: src/algorithm_cpp/problem226.cpp
class Solution {
public:
    TreeNode* invertTree(TreeNode* root) {
	if (root == nullptr) return nullptr;
	TreeNode* newleft = invertTree(root->right);
	TreeNode* newright = invertTree(root->left);
	root->left = newleft;
	root->right = newright;
	return orrt;
    }
};
